contributing.md

```diff
@@ -0,0 +1,135 @@
+      uri: https://github.com/trunk-io/plugins
+      ref: v0.0.8
+```
+
+Plugins can also be loaded from local paths, as below:
+
+```yaml
+plugins:
+  sources:
+    - id: trunk
+      local: /path/to/repo/root
+```
+
+Run `trunk run toggle-local` to quickly toggle this setting.
+
+Adding a plugin source lets users run `trunk check enable` or `trunk actions enable` with linters
+and actions defined in that plugin. For more information, see our
+[docs](https://docs.trunk.io/docs/plugins).
+
+If you want to request a new feature or get help with a problem, please see our
+[Features Page](https://features.trunk.io/) or our [Slack](https://slack.trunk.io/).
+
+## Releases
+
+Plugin sources are tied to the version specified with the `ref` field. When users run
+`trunk upgrade`, 3 fields in their trunk.yaml will be updated:
+
+1. The CLI version will be updated to the latest trunk binary.
+2. The plugin `ref` will be updated to the latest compatible release of this repo.
+3. All enabled linters will be updated to their latest versions.
+
+We aim to provide a fairly regular release cadence for this repository. Releases are tied to minimum
+compatible versions of the trunk CLI, identified by the `required_trunk_version` field. Users will
+not be able to load a plugin source until they have upgraded to a sufficient CLI version.
+
+If you need a linter or action definition from this repo that hasn't been released yet, you can
+change the `ref` field of a plugin source to be any `git` ref. However, we recommend sticking to our
+official tagged releases unless absolutely necessary.
+
+## Linters
+
+If you want to add a new linter integration with trunk, refer to our docs on
+[custom linters](https://docs.trunk.io/docs/check-custom-linters) and
+[custom parsers](https://docs.trunk.io/docs/custom-parsers), and work through the following:
+
+1. Create a directory inside `linters/` with the name of your new linter.
+2. Inside this new directory, create the following structure. If you ran `trunk check` in this
+   repository recently, some of these files should be automatically created for you:
+
+   ```text
+   linters/
+   └─my-linter/
+     │ plugin.yaml
+     │ my_linter.test.ts
+     │ readme.md (optional)
+     │ my-config.json (optional)
+     └─test_data/
+        └─basic.in.py (with appropriate extension)
+   ```
+
+3. Modify plugin.yaml to define your new linter. Refer to other linters in this repository as
+   examples.
+4. Making sure the plugin in `.trunk/trunk.yaml` is pointing to your local repository, run
+   `trunk check enable <my-linter>` to enable your linter, and run `trunk check` to verify that the
+   configuration is valid and that you get desired diagnostics. Running `trunk check --verbose` can
+   help provide greater insights when debugging.
+5. Add a few simple test cases to `my_linter.test.ts` to exercise your linter and generate
+   snapshots. Refer to [Testing Guidelines](tests/readme.md) for more information on writing and
```

Comment:
Do we need to document `linter-test-helper` somewhere?

Reply:
Not sure on this point. I made a note about files should be created for you, but I think it's more of a nice-to-have and not really a robust tool we want to advertise too overtly? Open to thoughts.

Reply:
Post-discussion. No need to document now, until more robust